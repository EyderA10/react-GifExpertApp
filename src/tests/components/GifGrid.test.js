import React from 'react';
import { useFetchGifs } from '../../Hooks/useFetchGifs';
const { shallow } = require("enzyme")
const { default: GifGrid } = require("../../components/GifGrid")
jest.mock('../../Hooks/useFetchGifs');

describe('Pruebas con gif grid', () => {

    const category = 'Hola mundo';

    test('Se debe mostrar correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(true);
        expect(wrapper.find('GifGridItem').length).toBe(0);
    })

    test('debe de mostrar items cuando se carga imagenes useFetch', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://urlc.af.com',
            title: 'titluos'
        },
        {
            id: '123',
            url: 'https://urlc.af.com',
            title: 'titluos'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })
    
    
})
