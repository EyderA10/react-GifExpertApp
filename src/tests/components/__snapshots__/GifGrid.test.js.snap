// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con gif grid Se debe mostrar correctamente 1`] = `
<div>
  <h3
    className="animate__animated animate__delay-2s 2s"
  >
    Hola mundo
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando Gifs..
  </p>
  <div
    className="card-grid"
  />
</div>
`;
